tests/ga/test_remoteaccess_handler.py

```diff
@@ -58,478 +82,431 @@ def setUp(self):
         # Since ProtocolUtil is a singleton per thread, we need to clear it to ensure that the test cases do not
         # reuse a previous state
         clear_singleton_instances(ProtocolUtil)
-        del info_messages[:]
-        del error_messages[:]
         for data in TestRemoteAccessHandler.eventing_data:
             del data
 
     # add_user tests
-    @patch('azurelinuxagent.common.logger.Logger.info', side_effect=log_info)
-    @patch('azurelinuxagent.common.logger.Logger.error', side_effect=log_error)
     @patch('azurelinuxagent.common.utils.cryptutil.CryptUtil.decrypt_secret', return_value="]aPPEv}uNg1FPnl?")
-    def test_add_user(self, _1, _2, _3):
-        rah = RemoteAccessHandler(Mock())
-        rah.os_util = MockOSUtil()
-        tstpassword = "]aPPEv}uNg1FPnl?"
-        tstuser = "foobar"
-        expiration_date = datetime.utcnow() + timedelta(days=1)
-        pwd = tstpassword
-        rah.add_user(tstuser, pwd, expiration_date)
-        users = get_user_dictionary(rah.os_util.get_users())
-        self.assertTrue(tstuser in users, "{0} missing from users".format(tstuser))
-        actual_user = users[tstuser]
-        expected_expiration = (expiration_date + timedelta(days=1)).strftime("%Y-%m-%d")
-        self.assertEqual(actual_user[7], expected_expiration)
-        self.assertEqual(actual_user[4], "JIT_Account")
-        self.assertEqual(0, len(error_messages))
-        self.assertEqual(1, len(info_messages))
-        self.assertEqual(info_messages[0], "User '{0}' added successfully with expiration in {1}"
-                         .format(tstuser, expected_expiration))
-
-    @patch('azurelinuxagent.common.logger.Logger.info', side_effect=log_info)
-    @patch('azurelinuxagent.common.logger.Logger.error', side_effect=log_error)
+    def test_add_user(self, *_):
+        with patch("azurelinuxagent.ga.remoteaccess.get_osutil", return_value=MockOSUtil()):
+            rah = RemoteAccessHandler(Mock())
+            tstpassword = "]aPPEv}uNg1FPnl?"
+            tstuser = "foobar"
+            expiration_date = datetime.utcnow() + timedelta(days=1)
+            pwd = tstpassword
+            rah._add_user(tstuser, pwd, expiration_date)
```

Comment:
Probably (definitely) out of the scope of this PR, but this should ideally call "run()" instead of a private method (lines 97, 114, 126, etc...). Is it worth capturing this as a future task, or should we just make the changes if these tests break?

Reply:
Yes, most of those tests need rewriting and unfortunately that is very common across most of the code base so opening tasks could be unmanageable. I tend to do improvements as I touch related code, many times I span the improvements over several iterations of the code.

Reply:
Of course, if a test breaks while changing the prod code we need to address that immediately. When that happens to you, try to evaluate whether the broken test adds any value, on several occasions I've found fixing broken tests is not worth the effort (they may be redundant, test private implementation, etc)